crates/ursa-pod/benches/e2e.rs

```diff
@@ -0,0 +1,271 @@
+use criterion::{measurement::Measurement, *};
+use futures::Future;
+use std::time::Duration;
+use tokio::sync::oneshot;
+
+const MAX_REQUESTS: usize = 64;
+
+const KILOBYTE_FILES: &[&[u8]] = &[
+    &[0u8; 1024],
+    &[0u8; 2 * 1024],
+    &[0u8; 4 * 1024],
+    &[0u8; 8 * 1024],
+    &[0u8; 16 * 1024],
+    &[0u8; 32 * 1024],
+    &[0u8; 64 * 1024],
+    &[0u8; 128 * 1024],
+    &[0u8; 256 * 1024],
+    &[0u8; 512 * 1024],
+];
+const MEGABYTE_FILES: &[&[u8]] = &[
+    &[0u8; 1024 * 1024],
+    &[0u8; 2 * 1024 * 1024],
+    &[0u8; 4 * 1024 * 1024],
+    &[0u8; 8 * 1024 * 1024],
+    &[0u8; 16 * 1024 * 1024],
+    &[0u8; 32 * 1024 * 1024],
+    &[0u8; 64 * 1024 * 1024],
+    &[0u8; 128 * 1024 * 1024],
+    &[0u8; 256 * 1024 * 1024],
+    &[0u8; 512 * 1024 * 1024],
+];
+
+fn benchmark_sizes<T: Measurement, C, S>(
+    g: &mut BenchmarkGroup<T>,
+    files: &[&'static [u8]],
+    unit: usize,
+    client: impl Fn(String, usize) -> C,
+    server: impl Fn(String, &'static [u8], oneshot::Sender<u16>) -> S,
+) where
+    C: Future,
+    S: Future + Send + 'static,
+    S::Output: Send + 'static,
+{
+    let runtime = tokio::runtime::Builder::new_multi_thread()
+        .enable_all()
+        .build()
+        .unwrap();
+
+    for file in files {
+        // Spawn the server and wait for it to signal that it's ready.
+        let (tx_started, rx_started) = oneshot::channel();
+        let server_task = runtime.spawn(server("127.0.0.1:0".into(), file, tx_started));
+        let port = futures::executor::block_on(rx_started).unwrap();
+        let addr = format!("127.0.0.1:{port}");
+
+        let mut num_requests = 1;
+        while num_requests <= MAX_REQUESTS {
+            let len = file.len() * num_requests;
+            g.throughput(Throughput::Bytes(len as u64));
+
+            // We need to allocate additional time to carry the same accuracy between the benchmarks
+            let mut time = Duration::from_secs(10 + num_requests as u64);
+            time += Duration::from_micros(len as u64 / 40);
+            g.measurement_time(time);
+
+            g.bench_with_input(
+                BenchmarkId::new(
+                    format!(
+                        "{num_requests} request{}",
+                        if num_requests != 1 { "s" } else { "" }
+                    ),
+                    file.len() / unit,
+                ),
+                &num_requests,
+                |b, &n| {
+                    b.to_async(&runtime).iter(|| client(addr.clone(), n));
+                },
+            );
+
+            num_requests *= 2;
+        }
+
+        server_task.abort();
+    }
+}
+
+fn protocol_benchmarks(c: &mut Criterion) {
+    // benchmark different file sizes
+    for (range, files, unit) in [
+        ("Content Size (Kilobyte)", KILOBYTE_FILES, 1024),
+        ("Content Size (Megabyte)", MEGABYTE_FILES, 1024 * 1024),
+    ] {
+        {
```

Comment:
Why remove the cfg?

Reply:
basically the default behavior will now run all benchmarks instead of just one protocol at a time. For the previous behavior (one at a time), you can use the optional regex string cargo bench accepts